skyportal/handlers/api/tag.py

```diff
@@ -167,34 +151,47 @@ def post(self):
             ).first():
                 return self.error("Specified tag does not exist", status=404)
 
-            # Verify source exists
+            # Verify obj exists
             if not session.scalars(
-                Source.select(session.user_or_token).where(Source.id == source_id)
+                Obj.select(session.user_or_token).where(Obj.id == obj_id)
             ).first():
-                return self.error("Specified source does not exist", status=404)
+                return self.error("Specified obj does not exist", status=404)
+
+            if hasattr(self.current_user, "created_by"):
+                user_id = self.current_user.created_by.id
+            else:
+                user_id = self.current_user.id
 
-            new_assoc = ObjTags(objtagoption_id=objtagoption_id, source_id=source_id)
+            new_assoc = ObjTags(
+                objtagoption_id=objtagoption_id, obj_id=obj_id, user_id=user_id
+            )
             session.add(new_assoc)
             session.commit()
 
             return self.success(
```

Comment:
I've put that comment in the previous review,  but I see you create the `data` in a lot of handlers when you could just send the python object from the db (e.g. `new_assoc`) which gets auto converted to dict/JSON.